pkg/sql/contention/event_store_test.go

```diff
@@ -249,6 +254,114 @@ func BenchmarkEventStoreIntake(b *testing.B) {
 	}
 }
 
+func TestLogResolvedEvents(t *testing.T) {
+	ctx := context.Background()
+
+	// Set up log interception to capture structured AggregatedContentionInfo events
+	spy := logtestutils.NewStructuredLogSpy(
+		t,
+		[]logpb.Channel{logpb.Channel_SQL_EXEC},
+		[]string{"aggregated_contention_info"},
+		logtestutils.FromLogEntry[eventpb.AggregatedContentionInfo],
+	)
+	cleanup := log.InterceptWith(ctx, spy)
+	defer cleanup()
+
+	fingerprintIDToHex := func(id uint64) string {
+		return "‹\\x" + sqlstatsutil.EncodeStmtFingerprintIDToString(appstatspb.StmtFingerprintID(id)) + "›"
+	}
+	txnFingerprintIDToHex := func(id uint64) string {
+		return "‹\\x" + sqlstatsutil.EncodeTxnFingerprintIDToString(appstatspb.TransactionFingerprintID(id)) + "›"
+	}
+
+	// Create test contention events
+	events := []contentionpb.ExtendedContentionEvent{
+		// Two events with same waiting (stmt, txn), blocking txn and key
+		// These should aggregate.
+		{
+			WaitingStmtFingerprintID: 123,
+			WaitingTxnFingerprintID:  456,
+			BlockingTxnFingerprintID: 789,
+			BlockingEvent: kvpb.ContentionEvent{
+				Key:      []byte("test-key-1"),
+				Duration: 100 * time.Millisecond,
+			},
+		},
+		{
+			WaitingStmtFingerprintID: 123,
+			WaitingTxnFingerprintID:  456,
+			BlockingTxnFingerprintID: 789,
+			BlockingEvent: kvpb.ContentionEvent{
+				Key:      []byte("test-key-1"),
+				Duration: 200 * time.Millisecond,
+			},
+		},
+		// Key changes.
+		// New combination.
+		{
+			WaitingStmtFingerprintID: 123,
+			WaitingTxnFingerprintID:  456,
+			BlockingTxnFingerprintID: 789,
+			BlockingEvent: kvpb.ContentionEvent{
+				Key:      []byte("test-key-2"),
+				Duration: 100 * time.Millisecond,
+			},
+		},
+		// test-key-1 again, but different waiting stmt.
+		// New combination.
+		{
+			WaitingStmtFingerprintID: 321,
+			WaitingTxnFingerprintID:  456,
+			BlockingTxnFingerprintID: 789,
+			BlockingEvent: kvpb.ContentionEvent{
+				Key:      []byte("test-key-1"),
+				Duration: 50 * time.Millisecond,
+			},
+		},
+		// test-key-1 again, same waiting stmt and txn, but different blocking txn.
+		// New combination.
+		{
+			WaitingStmtFingerprintID: 123,
+			WaitingTxnFingerprintID:  456,
+			BlockingTxnFingerprintID: 888,
+			BlockingEvent: kvpb.ContentionEvent{
+				Key:      []byte("test-key-1"),
+				Duration: 20 * time.Millisecond,
+			},
+		},
+	}
+
+	logResolvedEvents(ctx, events)
+	contentionLogs := spy.GetLogs(logpb.Channel_SQL_EXEC)
+	require.Len(t, contentionLogs, 4)
+	for _, log := range contentionLogs {
```

Comment:
the `events` above are more like test cases, I would construct the events and expectations from the cases so you don't have to do a conditional like this in the test.

instead it's nicer when you iterate through each case and check if the `contentionLogs` contain a match.